src/common/k8s-api/api-manager/manager.injectable.ts

```diff
@@ -2,16 +2,25 @@
  * Copyright (c) OpenLens Authors. All rights reserved.
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
-import { getInjectable } from "@ogre-tools/injectable";
-import { apiManager } from "./legacy-global";
+import { getInjectable, getInjectionToken } from "@ogre-tools/injectable";
+import type { KubeObjectStore } from "../kube-object.store";
+import { ApiManager } from "./api-manager";
+
+export const kubeObjectStoreInjectionToken = getInjectionToken<KubeObjectStore<any, any, any>>({
+  id: "kube-object-store-token",
+});
 
 const apiManagerInjectable = getInjectable({
   id: "api-manager",
-  instantiate: () => {
-    const a = apiManager;
+  instantiate: (di) => {
+    const apiManager = new ApiManager();
+    const stores = di.injectMany(kubeObjectStoreInjectionToken);
+
+    for (const store of stores) {
+      apiManager.registerStore(store);
+    }
```

Comment:
This operational code should be something the `ApiManager` does itself, and not something the unrelated responsibility of instantiation does.

Reply:
You mean pass in the `KubeObjectStore[]` itself to the constructor?

Reply:
Yes, somewhat.

Is the misbehaving of extension-API preventing this again? If so, resolve -_-

Reply:
What is the benefit of having it in the constructor?

Reply:
The constructor would make the stores just another dependency with clarity of rules.

...However, another more interesting approach would be to do that **Hollywood-principle** thing, and not require registration at all, and instead call for all stores from within `ApiManager` when required.